make-all.sh

```diff
@@ -80,28 +93,42 @@ print_help() {
 
 print_cmake_params() {
     echo ""
-    [ -n "${_GENERATOR}" ] && echo "${_GENERATOR}"
-    [ -n "${_BUILD_TYPE}" ] && echo "${_BUILD_TYPE:2}"
-    echo "CMAKE_INSTALL_PREFIX=${_PREFIX}"
-    [ -n "${_STAGING_PREFIX}" ] && echo "${_STAGING_PREFIX:2}"
-    [ -n "${_TOOLCHAIN_FILE}" ] && echo "${_TOOLCHAIN_FILE:2}"
+    echo "Networking-recipe options:"
+    [ -n "${_BUILD_TYPE}" ] && echo "  ${_BUILD_TYPE:2}"
+    echo "  CMAKE_INSTALL_PREFIX=${_PREFIX}"
+    [ -n "${_COVERAGE}" ] && echo "  ${_COVERAGE:2}"
     [ -n "${_CXX_STD}" ] && echo "CMAKE_CXX_STANDARD=${_CXX_STD}"
-    echo "DEPEND_INSTALL_DIR=${_DEPS_DIR}"
-    [ -n "${_HOST_DEPEND_DIR}" ] && echo "${_HOST_DEPEND_DIR:2}"
-    echo "OVS_INSTALL_DIR=${_OVS_DIR}"
-    echo "SDE_INSTALL_DIR=${_SDE_DIR}"
-    [ -n "${_WITH_KRNLMON}" ] && echo "${_WITH_KRNLMON:2}"
-    [ -n "${_WITH_OVSP4RT}" ] && echo "${_WITH_OVSP4RT:2}"
-    [ -n "${_COVERAGE}" ] && echo "${_COVERAGE:2}"
-    echo "${_SET_RPATH:2}"
-    echo "${_TARGET_TYPE:2}"
-    if [ ${_DO_BUILD} -eq 0 ]; then
+    echo "  DEPEND_INSTALL_DIR=${_DEPS_DIR}"
+    [ -n "${_GENERATOR}" ] && echo "  ${_GENERATOR}"
+    [ -n "${_HOST_DEPEND_DIR}" ] && echo "  ${_HOST_DEPEND_DIR:2}"
+    echo "  JOBS=${_NJOBS}"
+    echo "  OVS_INSTALL_DIR=${_OVS_DIR}"
+    [ -n "${_P4OVS_MODE}" ] && echo "  ${_P4OVS_MODE:2}"
+    echo "  SDE_INSTALL_DIR=${_SDE_DIR}"
+    echo "  ${_SET_RPATH:2}"
+    [ -n "${_STAGING_PREFIX}" ] && echo "  ${_STAGING_PREFIX:2}"
+    echo "  ${_TARGET_TYPE:2}"
+    [ -n "${_TOOLCHAIN_FILE}" ] && echo "  ${_TOOLCHAIN_FILE:2}"
+    [ -n "${_WITH_KRNLMON}" ] && echo "  ${_WITH_KRNLMON:2}"
+    [ -n "${_WITH_OVSP4RT}" ] && echo "  ${_WITH_OVSP4RT:2}"
+
+    echo ""
+    echo "OvS options:"
+    [ -n "${_P4OVS_MODE}" ] && echo "  ${_P4OVS_MODE:2}"
+    [ -n "${_PKG_CONFIG_PATH}" ] && echo "  PKG_CONFIG_PATH=${_PKG_CONFIG_PATH}"
+    if [ ${_ovs_first} -ne 0 ]; then
```

Comment:
Curious why we want this control of building OVS first or last. Is this for the ovs-p4rt dependency or somethign else?

Reply:
The immediate problem is that I need OvS to build `ovs-vswitchd` and `ovs-testcontroller` itself, against the `ovsp4rt_stubs` library, so I can run the OVS test suite against the P4 version of OVS.

The less immediate (but more important) problem is that the current build order is fundamentally wrong. networking-recipe doesn't depend on OVS; OVS depends on ovs-p4rt, which is built as part of networking-recipe.

I goofed two years ago, and am trying to put things right before upstreaming OVS.

For more information, see https://github.com/ipdk-io/networking-recipe/issues/516.